src/pudl/transform/ferc714.py

```diff
@@ -533,3 +541,40 @@ def out_ferc714__hourly_planning_area_demand(
         df[columns], table_name="out_ferc714__hourly_planning_area_demand"
     )
     return df
+
+
+@asset(
+    io_manager_key="parquet_io_manager",
+    op_tags={"memory-use": "high"},  # Should this be high?
+    compute_kind="pandas",
+)
+def out_ferc714__yearly_planning_area_forecast_demand(
+    raw_ferc714__yearly_planning_area_forecast_demand: pd.DataFrame,
```

Comment:
* For naming consistency I think it might be better to change both of these asset names to use `*_demand_forecast` rather than `*_forecast_demand`. We've already got a `demand_forecast_mwh` column in the EIA-930 data, and this naming would turn `_forecast` into a modifying suffix relative to `*_ferc714_hourly_planning_area_demand` instead of inserting the word into the middle of the analogous name.
* I think the new table that we're creating here is a `core_ferc714__` table, not an output table, since we're (yet) merging in any additional information to denormalize it and make it more human readable (e.g. the name of the planning area associated with the ID)

Reply:
Good call. Changing the suffix now. And I missed that part about `core` vs `out`. Thanks for catching that. Updating now as well.